src/python/pants/backend/python/typecheck/pyright/rules.py

```diff
@@ -110,17 +185,22 @@ async def pyright_typecheck_partition(
         ),
     )
 
-    # venv workaround as per: https://github.com/microsoft/pyright/issues/4051
-    dummy_config_digest = await Get(
-        Digest,
-        CreateDigest(
-            [
-                FileContent(
-                    "pyrightconfig.json",
-                    f'{{ "venv": "{requirements_venv_pex.venv_rel_dir}" }}'.encode(),
-                )
-            ]
-        ),
+    source_roots_get = Get(
+        SourceRootsResult,
+        SourceRootsRequest,
+        SourceRootsRequest.for_files(root_sources.snapshot.files),
+    )
+
+    requirements_venv_pex, source_roots = await MultiGet(
+        requirements_venv_pex_get,
+        source_roots_get,
+    )
+
+    # Patch the config file to use the venv directory from the requirements pex,
```

Comment:
Note: This might be temporary until we support Pyright's execution environments (which lets Pyright use multiple roots and a different config per root).

I think this is a better idea overall, if they can be distinguished.

Reply:
https://github.com/pantsbuild/pants/issues/17816